### PR TITLE
Support mysql tracing through the plugin API

### PR DESCRIPTION
**Pull request by DominicKramer**

### CODE REVIEW

**Comment by DominicKramer** [comment]
PTAL

File: src/plugins/plugin-mysql.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    var labels = {};
+    if (api.enhancedDatabaseReportingEnabled()) {
+      if (err) {
+        labels.error = err;
+      }
+      if (res) {
+        labels.result = res;
+      }
+    }
+    span.endSpan(labels);
```
Comment (kjin) [inline src/plugins/plugin-mysql.js]: `endSpan` currently takes no arguments, so you'll have to use `addLabel`. Do you think having `endSpan` take in an object with key-value fields is useful? (Here I think it's actually slightly cleaner to use `addLabel` on lines 53 and 56.)

Comment (DominicKramer) [inline src/plugins/plugin-mysql.js]: Done

File: src/plugins/plugin-mysql.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+var SUPPORTED_VERSIONS = '^2.9.x';
+
+function createQueryWrap(api, createQuery) {
+  return function createQuery_trace(sql, values, cb) {
+    var span = api.createChildSpan({
+      name: 'mysql-query'
+    });
+    var query = createQuery.apply(this, arguments);
+    if (span) {
```
Comment (matthewloring) [inline src/plugins/plugin-mysql.js]: Can we change this to an `if (!span) return query`?

**Pull request merged**